packages/framework/client-logger/app-insights-logger/src/test/fluidAppInsightsLogger.spec.ts

```diff
@@ -70,89 +70,29 @@ describe("FluidAppInsightsLogger", () => {
 				filters: [
 					{
 						namespacePattern: "A:B:C",
-						namespacePatternExceptions: ["D:C:A"],
+						namespacePatternExceptions: new Set(["D:C:A"]),
 					},
 				],
 			},
 		};
 		assert.throws(() => new FluidAppInsightsLogger(appInsightsClient, invalidConfig));
 	});
 
-	it("constructor() merges redundant pure category filters", () => {
+	it("constructor() throws error when multiple filters that only define categories are provided", () => {
 		const invalidConfig: FluidAppInsightsLoggerConfig = {
 			filtering: {
 				mode: "inclusive",
 				filters: [
-					{
-						categories: ["error", "generic"],
-					},
 					{
 						categories: ["error"],
 					},
 					{
-						categories: ["performance"],
+						categories: ["generic", "performance"],
 					},
 				],
 			},
 		};
-		const logger = new FluidAppInsightsLogger(appInsightsClient, invalidConfig);
-		assert.strictEqual(logger.filters.length, 1);
-		if ("categories" in logger.filters[0]) {
-			assert.strictEqual(true, logger.filters[0].categories.includes("error"));
-			assert.strictEqual(true, logger.filters[0].categories.includes("generic"));
-			assert.strictEqual(true, logger.filters[0].categories.includes("performance"));
-		} else {
-			// The merged category filter should be a CateogryFilter and have the "categories" attribute
-			fail();
-		}
-	});
-
-	it("sortTelemetryFilters() orders filters from longest namespace to shortest", () => {
-		const expectedSortedFilterArray: TelemetryFilter[] = [
-			{
-				namespacePattern: "veryLongNamespacePattern333",
-				categories: ["performance"],
-			},
-			{
-				namespacePattern: "veryLongNamespacePattern22",
-				categories: ["performance", "generic"],
-			},
-			{
-				namespacePattern: "veryLongNamespacePattern1",
-			},
-			{
-				namespacePattern: "namespace1",
-				categories: ["performance"],
-			},
-			{
-				namespacePattern: "namespace",
-				categories: ["performance", "generic"],
-			},
-			{
-				namespacePattern: "short",
-			},
-			{
-				categories: ["performance"],
-			},
-		];
-
-		const unsortedFilters: TelemetryFilter[] = [
-			expectedSortedFilterArray[2],
-			expectedSortedFilterArray[4],
-			expectedSortedFilterArray[6],
-			expectedSortedFilterArray[1],
-			expectedSortedFilterArray[3],
-			expectedSortedFilterArray[0],
-			expectedSortedFilterArray[5],
-		];
-
-		const logger = new FluidAppInsightsLogger(appInsightsClient, {
-			filtering: { mode: "exclusive", filters: unsortedFilters },
-		});
-
-		debugger;
-
-		assert.deepStrictEqual(logger.filters, expectedSortedFilterArray);
+		assert.throws(() => new FluidAppInsightsLogger(appInsightsClient, invalidConfig));
```

Comment:
Nit (let's create a follow up task for it): when asserting that something throws, it's a good practice to check that the thrown error is what we expect. Otherwise we might introduce a change in behavior that causes a different problem, and we wouldn't know it with this test.

Reply:
I've gone ahead and updated the tests to assert for a specific error and message combination. In the future we may want to consider standardizing the error messages